arkouda/series.py

```diff
@@ -419,70 +422,181 @@ def to_dataframe(
+            a fluid programming style.
+            Please note you cannot register two different Series with the same name.
+
+        Raises
+        ------
+        TypeError
+            Raised if user_defined_name is not a str
+        RegistrationError
+            If the server was unable to register the Series with the user_defined_name
+
+        See also
+        --------
+        unregister, attach, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
         """
+        from arkouda.client import generic_msg
+
+        if self.registered_name is not None and self.is_registered():
+            raise RegistrationError(f"This object is already registered as {self.registered_name}")
+        generic_msg(
+            cmd="register",
+            args={
+                "name": user_defined_name,
+                "objType": self.objType,
+                "num_idxs": 1,
+                "idx_names": [
+                    json.dumps(
+                        {
+                            "codes": self.index.values.codes.name,
+                            "categories": self.index.values.categories.name,
+                            "NA_codes": self.index.values._akNAcode.name,
+                            **(
+                                {"permutation": self.index.values.permutation.name}
+                                if self.index.values.permutation is not None
+                                else {}
+                            ),
+                            **(
+                                {"segments": self.index.values.segments.name}
+                                if self.index.values.segments is not None
+                                else {}
+                            ),
+                        }
+                    )
+                    if isinstance(self.index.values, Categorical)
+                    else self.index.values.name
+                ],
+                "idx_types": [self.index.values.objType],
+                "values": json.dumps(
+                    {
+                        "codes": self.values.codes.name,
+                        "categories": self.values.categories.name,
+                        "NA_codes": self.values._akNAcode.name,
+                        **(
+                            {"permutation": self.values.permutation.name}
+                            if self.values.permutation is not None
+                            else {}
+                        ),
+                        **(
+                            {"segments": self.values.segments.name}
+                            if self.values.segments is not None
+                            else {}
+                        ),
+                    }
+                )
+                if isinstance(self.values, Categorical)
+                else self.values.name,
+                "val_type": self.values.objType,
```

Comment:
This is in the logic setting the keys to send the message to the server. Series will have a single value object (set as `val_type`. The `if isinstance(self.values, Categorical)` creates the JSON of the categorical components and otherwise sends the name of the object. `val_type` is the next key in the message JSON.

Reply:
ah yah I was misreading, thanks for the explaination!